lib/gcp/storage.js

```diff
@@ -23,88 +23,46 @@ function _getDefaultBucket(projectId) {
   });
 }
 
-// helper functions for creating staging buckets, delete later-----------
-function _getOperation(operation) {
-  return api.request('GET', '/v1/' + operation, {
-    auth: true,
-    origin: api.appengineOrigin
-  });
-}
-
-function _pollOperation(operation) {
-  var POLL_INTERVAL = 500; // 0.5 second
-  return new RSVP.Promise(function(resolve) {
-    setTimeout(function() { resolve(); }, POLL_INTERVAL);
-  }).then(function() {
-    return _getOperation(operation).then(function(res) {
-      if (res.body.done) {
-        return true;
-      }
-      return _pollOperation(operation);
-    });
-  });
-}
-// end of helper functions for creating staging buckets-------------------
-
 function _getBucket(bucket) {
   return api.request('GET', '/storage/' + version + '/b/' + bucket, {
     auth: true,
     origin: api.googleOrigin
   });
 }
 
-function _createBucket(projectId) {
-  // var bucket = _functionsBucketName(projectId);
-  // return api.request('POST', '/storage/' + version + '/b?project=' + projectId, {
-  //   auth: true,
-  //   data: {
-  //     name: bucket
-  //   },
-  //   origin: api.googleOrigin
-  return api.request('POST', '/v1/apps/' + projectId + ':repair', {
+function _getUploadUrl(projectId, location) {
+  var parent = 'projects/' + projectId + '/locations/' + location;
+  return api.request('POST', '/v1/' + parent + '/functions:generateUploadUrl', {
```

Comment:
because this is an API on cloud functions, it should be part of the functions library, not the GCS library.